From d493bf96fec9ec55c45f24de7b296142ffa0ccff Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgads@gmail.com>
Date: Mon, 3 Jun 2019 08:37:40 +0300
Subject: [PATCH] Fix cache dir resolving on Windows

... when not running from MSYS environment, and HOME is not set.
---
 gdb/common/pathstuff.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gdb/common/pathstuff.c b/gdb/common/pathstuff.c
index 2b1669a5b9..edb8a1fcda 100644
--- a/gdb/common/pathstuff.c
+++ b/gdb/common/pathstuff.c
@@ -231,6 +231,10 @@ get_standard_cache_dir ()
 #endif
 
   const char *home = getenv ("HOME");
+#ifdef _WIN32
+  if (home == nullptr)
+      home = getenv ("USERPROFILE");
+#endif
   if (home != NULL)
     {
       /* Make sure the path is absolute and tilde-expanded.  */
-- 
2.22.0.rc0.windows.1

